### PR TITLE
Added tests stubs for errata mgmt

### PR DESCRIPTION
**Pull request by sthirugn**
77 test stubs ready :

```py
# nosetests -c robottelo.properties tests.foreman.api.test_errata
@Test: Filter applicable errata for a content host by current and ... SKIP: Test not implemented
@Test: Available errata count when retrieving Content host ... SKIP: Test not implemented
@Test: Generate a difference in errata between a set of enviroments ... SKIP: Test not implemented
@Test: View all errata specific to an Org ... SKIP: Test not implemented
@Test: View all errata in an Org sorted by Updated ... SKIP: Test not implemented
@Test: Filter errata by CVE ... SKIP: Test not implemented
@Test: Filter errata by issued date ... SKIP: Test not implemented
@Test: View a list of affected content hosts for an erratum ... SKIP: Test not implemented
@Test: Filter errata list based on affected content hosts ... SKIP: Test not implemented
@Test: Install errata in a host-collection ... SKIP: Test not implemented
@Test: Select multiple errata and apply them to multiple content ... SKIP: Test not implemented
@Test: Query a subset of environments or content views to push new ... SKIP: Test not implemented
@Test: Select multiple packages and apply them to multiple content ... SKIP: Test not implemented

Ran 13 tests in 0.004s
OK (SKIP=13)

# nosetests -c robottelo.properties tests.foreman.cli.test_errata
@Test: Available errata count displayed while viewing a list of ... SKIP: Test not implemented
@Test: Filter errata by product id ... SKIP: Test not implemented
@Test: Filter errata by Org id ... SKIP: Test not implemented
@Test: Filter errata by Org name ... SKIP: Test not implemented
@Test: Filter errata by Org label ... SKIP: Test not implemented
@Test: Filter errata by CVE ... SKIP: Test not implemented
@Test: Filter errata by product name ... SKIP: Test not implemented
@Test: Filter errata by product id and Org id ... SKIP: Test not implemented
@Test: Filter errata by product id and Org name ... SKIP: Test not implemented
@Test: Filter errata by product id ... SKIP: Test not implemented
@Test: Filter errata by product name ... SKIP: Test not implemented
@Test: Filter errata by product name and Org id ... SKIP: Test not implemented
@Test: Filter errata by product name and Org name ... SKIP: Test not implemented
@Test: Filter errata by product name and Org label ... SKIP: Test not implemented
@Test: Show errata only if the User has permissions to view them ... SKIP: Test not implemented
@Test: Sort errata by Issued date ... SKIP: Test not implemented
@Test: Filter errata by org id and sort by updated date ... SKIP: Test not implemented
@Test: Filter errata by org name and sort by updated date ... SKIP: Test not implemented
@Test: Filter errata by org label and sort by updated date ... SKIP: Test not implemented
@Test: Filter errata by org id and sort by issued date ... SKIP: Test not implemented
@Test: Filter errata by org name and sort by issued date ... SKIP: Test not implemented
@Test: Filter errata by org label and sort by issued date ... SKIP: Test not implemented
@Test: View a list of affected content hosts for an erratum ... SKIP: Test not implemented
@Test: View a list of affected content hosts for an erratum filtered ... SKIP: Test not implemented
@Test: Using hc-id and org id to install an erratum in a hc ... SKIP: Test not implemented
@Test: Attempt to install an erratum in a hc using hc-id and not ... SKIP: Test not implemented
@Test: Attempt to install an erratum in a hc without specifying hc ... SKIP: Test not implemented
@Test: Using hc-id and org name to install an erratum in a hc ... SKIP: Test not implemented
@Test: Use hc-id and org label to install an erratum in a hc ... SKIP: Test not implemented
@Test: Use hc-name and org id to install an erratum in a hc ... SKIP: Test not implemented
@Test: Use hc name and org name to install an erratum in a hc ... SKIP: Test not implemented
@Test: Use hc-name and org label to install an erratum in a hc ... SKIP: Test not implemented
@Test: Attempt to install an erratum in a hc using hc-id and not ... SKIP: Test not implemented
@Test: Attempt to install an erratum in a hc using hc-name and not ... SKIP: Test not implemented
@Test: Attempt to install an erratum in a hc by not specifying hc ... SKIP: Test not implemented

Ran 35 tests in 0.007s
OK (SKIP=35)


# nosetests -c robottelo.properties tests.foreman.api.test_errata
@Test: Filter applicable errata for a content host by current and ... SKIP: Test not implemented
@Test: Available errata count when retrieving Content host ... SKIP: Test not implemented
@Test: Generate a difference in errata between a set of enviroments ... SKIP: Test not implemented
@Test: View all errata specific to an Org ... SKIP: Test not implemented
@Test: View all errata in an Org sorted by Updated ... SKIP: Test not implemented
@Test: Filter errata by CVE ... SKIP: Test not implemented
@Test: Filter errata by issued date ... SKIP: Test not implemented
@Test: View a list of affected content hosts for an erratum ... SKIP: Test not implemented
@Test: Filter errata list based on affected content hosts ... SKIP: Test not implemented
@Test: Install errata in a host-collection ... SKIP: Test not implemented
@Test: Select multiple errata and apply them to multiple content ... SKIP: Test not implemented
@Test: Query a subset of environments or content views to push new ... SKIP: Test not implemented
@Test: Select multiple packages and apply them to multiple content ... SKIP: Test not implemented

Ran 13 tests in 0.004s
OK (SKIP=13)

# nosetests -c robottelo.properties tests.foreman.ui.test_email
@Test: Receive daily email with host advisory information ... SKIP: Test not implemented
@Test: Receive weekly email with host advisory information ... SKIP: Test not implemented
@Test: Receive monthly email with host advisory information ... SKIP: Test not implemented
@Test: Receive no email with host advisory information ... SKIP: Test not implemented
@Test: Receive 'Katello Sync Errata' notifications - only for ... SKIP: Test not implemented
@Test: Receive 'Katello Promote Errata' notifications - only for ... SKIP: Test not implemented
@Test: Receive 'Katello Host Advisory' notifications - only for ... SKIP: Test not implemented
@Test: Receive 'Katello Host Advisory' notifications - only for ... SKIP: Test not implemented
@Test: Manage user email notification preferences ... SKIP: Test not implemented
@Test: Receive email after every promote operation ... SKIP: Test not implemented
@Test: Do not receive email after every promote operation ... SKIP: Test not implemented
@Test: Receive email after puppet error ... SKIP: Test not implemented
@Test: Do not receive email after puppet error ... SKIP: Test not implemented
@Test: Receive email after every sync operation ... SKIP: Test not implemented
@Test: Do not receive email after every sync operation ... SKIP: Test not implemented

Ran 15 tests in 0.005s
OK (SKIP=15)

# nosetests -c robottelo.properties tests.foreman.api.test_email
@Test: Manage user email notification preferences. ... SKIP: Test not implemented

Ran 1 test in 0.002s
OK (SKIP=1)

```

### CODE REVIEW

File: tests/foreman/api/test_email.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+"""API Tests for the email notification feature"""
+
+import sys
+
+if sys.hexversion >= 0x2070000:
+    import unittest
+else:
+    import unittest2 as unittest
```
Comment (Ichimonji10) [inline tests/foreman/api/test_email.py]: This works fine, but `sys.version_info.major` is a more precise tool:

```python
from sys import version_info
if version_info.major == 2:
    import unittest2 as unittest
else:
    import unittest
```

File: tests/foreman/api/test_errata.py [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+        """@Test: Filter applicable errata for a content host by current and
+        Library environments
+
+        @Feature: Errata
+
+        @Setup:
+
+        1. Make sure multiple environments are present.
+
+        2. One of Content host's previous environment has additional errata.
```
Comment (Ichimonji10) [inline tests/foreman/api/test_errata.py]: This works fine, but the docstring can be made a little more compact by dropping the whitespace between list items. See example [here](http://rst.ninjs.org/?n=4bb0882de29394dbfedda0a372be75dd&theme=basic).

File: tests/foreman/api/test_email.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+from robottelo.common.constants import NOT_IMPLEMENTED
+from robottelo.test import APITestCase
+
+
+class EmailTestCase(APITestCase):
+    """API Tests for the email notification feature"""
+
+    @unittest.skip(NOT_IMPLEMENTED)
```
Comment (Ichimonji10) [inline tests/foreman/api/test_email.py]: `robottelo.common.decorators.stubbed` can also be used. See documentation [here](http://robottelo.readthedocs.org/en/latest/api/robottelo.common.html#robottelo.common.decorators.stubbed) and an example of a module which uses it [here](https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/api/test_hostgroup.py).

**Comment by Ichimonji10** [comment]
ACK pending comments.

**Comment by sthirugn** [comment]
@Ichimonji10 ty, comments addressed now.

**Comment by JacobCallahan** [comment]
Wow! Impressive work. ACK

**Pull request merged**